src/osd/osd_types.cc

```diff
@@ -4125,9 +4120,13 @@ ostream& operator<<(ostream& out, const pg_log_entry_t& e)
 
 // -- pg_log_dup_t --
 
-string pg_log_dup_t::get_key_name() const
+std::string pg_log_dup_t::get_key_name() const
 {
-  return "dup_" + version.get_key_name();
+  alignas(4) static const char prefix[] = { 'd', 'u', 'p', '_' };
+  alignas(4) char key[36];
+  memcpy(key, prefix, 4);
+  version.get_key_name(key + 4);
+  return std::string(key);
```

Comment:
I noticed the opposite on cppreference:

(1) Modifying the character array accessed through the const overload of data has undefined behavior.

And up until C++17 it looks like we have:

`const CharT* data() const;` as the only overload

I may be wrong, though.

Could we use a string_view in the relevant places instead?